### DISCUSSION TITLE
Jobs sceduling on server that can be updated with a wrong system time

### DISCUSSION THREAD

**Discussion by oyemini-fortifieddata** [quartznet/quartznet#1138] [Q&A]
Hi all,

I've a scheduler application (base on Quartz of course) on one of our clients servers. at times, one particular server updates it system time to a few weeks in the future and then get back to the original time, after the system update the scehduler stops running jobs. Client IT guys tried to solve this for weeks with no success, so my question: How Quartz is dependant on system time? I know that for example, [C# timers](https://stackoverflow.com/questions/5128027/which-timers-are-dependent-on-system-time) are not based on the system time. How can I use it so that jobs sceduing won't be affected? should I stick with simple trigger instead of a cron trigger? Do the library is so dependent on system time and I'll have to find a different solution?

**Comment by lahma** [comment]
Quartz calculates next fire time ahead of the fire and it's based on `DateTme.UtcNow` calculations, which allows Quartz to work between restarts and in clustered mode - hardware clocks are not suitable when scheduling can be distributed - we need precise UTC instants.

So Quartz is not suitable to be run on such unstable non-production-ready systems. So if the system time isn't stable Quartz won't work there. So maybe `Timer`s could work, but seems that there are more dire problems in the IT infrastructure anyways...
